File: components/content_capture/android/java/src/org/chromium/components/content_capture/ContentCaptureFeatures.java
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
package org.chromium.components.content_capture;

import org.chromium.base.CommandLine;

/**
 * The class to get if feature is enabled from native.
 */
public class ContentCaptureFeatures {
    private static final String FLAG = "dump-captured-content-to-logcat-for-testing";

    public static boolean isEnabled() {
        return nativeIsEnabled();
    }

    public static boolean isDumpForTestingEnabled() {
        return CommandLine.getInstance().hasSwitch(FLAG);
    }
    private static native boolean nativeIsEnabled();
}

File: components/content_capture/common/content_capture_features.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2019 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/content_capture/common/content_capture_features.h"

#include "build/build_config.h"

namespace content_capture::features {

constexpr int kTaskInitialDelayMs = 500;

bool IsContentCaptureEnabled() {
#if BUILDFLAG(IS_ANDROID)
  return true;
#else
  return false;
#endif
}

int TaskInitialDelayInMilliseconds() {
  return kTaskInitialDelayMs;
}

}  // namespace content_capture::features
